src/core/server/saved_objects/migrationsv2/integration_tests/multiple_kibana_nodes.test.ts

```diff
@@ -0,0 +1,219 @@
+
+async function removeLogFiles() {
+  glob(Path.join(__dirname, `${LOG_FILE_PREFIX}_*.log`), (err, files) => {
+    files.forEach(async (file) => {
+      // ignore errors if it doesn't exist
+      await asyncUnlink(file).catch(() => void 0);
+    });
+  });
+}
+
+function extractSortNumberFromId(id: string): number {
+  const parsedId = parseInt(id.split(':')[1], 10); // "foo:123" -> 123
+  if (isNaN(parsedId)) {
+    throw new Error(`Failed to parse Saved Object ID [${id}]. Result is NaN`);
+  }
+  return parsedId;
+}
+
+async function fetchDocs(esClient: ElasticsearchClient, index: string) {
+  const { body } = await esClient.search<any>({
+    index,
+    size: 10000,
+    body: {
+      query: {
+        bool: {
+          should: [
+            {
+              term: { type: 'foo' },
+            },
+          ],
+        },
+      },
+    },
+  });
+
+  return body.hits.hits
+    .map((h) => ({
+      ...h._source,
+      id: h._id,
+    }))
+    .sort((a, b) => extractSortNumberFromId(a.id) - extractSortNumberFromId(b.id));
+}
+
+interface CreateRootConfig {
+  logFileName: string;
+}
+
+function createRoot({ logFileName }: CreateRootConfig) {
+  return kbnTestServer.createRoot(
+    {
+      elasticsearch: {
+        hosts: [esTestConfig.getUrl()],
+        username: kibanaServerTestUser.username,
+        password: kibanaServerTestUser.password,
+      },
+      migrations: {
+        skip: false,
+        enableV2: true,
+        batchSize: 100, // fixture contains 5000 docs
+      },
+      logging: {
+        appenders: {
+          file: {
+            type: 'file',
+            fileName: logFileName,
+            layout: {
+              type: 'pattern',
+            },
+          },
+        },
+        loggers: [
+          {
+            name: 'root',
+            appenders: ['file'],
+          },
+          {
+            name: 'savedobjects-service',
+            appenders: ['file'],
+            level: 'debug',
+          },
+        ],
+      },
+    },
+    {
+      oss: true,
+    }
+  );
+}
+
+describe('migration v2', () => {
+  let esServer: kbnTestServer.TestElasticsearchUtils;
+  let rootA: Root;
+  let rootB: Root;
+  let rootC: Root;
+
+  const migratedIndex = `.kibana_${pkg.version}_001`;
+  const fooType: SavedObjectsType = {
+    name: 'foo',
+    hidden: false,
+    mappings: { properties: { status: { type: 'text' } } },
+    namespaceType: 'agnostic',
+    migrations: {
+      '7.14.0': (doc) => {
+        if (doc.attributes?.status) {
+          doc.attributes.status = doc.attributes.status.replace('not_migrated', 'migrated');
+        }
+        return doc;
+      },
+    },
+  };
+
+  beforeAll(async () => {
+    await removeLogFiles();
+  });
+
+  afterAll(async () => {
+    await new Promise((resolve) => setTimeout(resolve, 10000));
+  });
+
+  beforeEach(async () => {
+    rootA = createRoot({
+      logFileName: Path.join(__dirname, `${LOG_FILE_PREFIX}_A.log`),
+    });
```

Comment:
At first I took a more fancy approach and started writing a helper that could manage the setup/teardown of multiple `Root`s, but then just stayed with this implementation as it was a bit more explicit and easier to follow.

Perhaps if we find ourselves adding more integration tests for multi-Kibana scenarios in the future, we can add something more generic into `test_helpers`